core/src/transport/mod.rs

```diff
@@ -116,8 +116,12 @@ pub trait Transport {
     /// a remote node observes for one of our dialers.
     ///
     /// For example, if `server` is `/ip4/0.0.0.0/tcp/3000` and `observed` is
-    /// `/ip4/80.81.82.83/tcp/29601`, then we should return `/ip4/80.81.82.83/tcp/3000`. Each
-    /// implementation of `Transport` is only responsible for handling the protocols it supports.
+    /// `/ip4/80.81.82.83/tcp/29601`, then we should return `/ip4/80.81.82.83/tcp/3000`.
+    ///
+    /// Each implementation of `Transport` is only responsible for handling the protocols it
```

Comment:
What I'd like to see explained is that the implementation should assume that `server` matches the protocol the `Transport` handles, but `observed` can be anything.

Reply:
See [my other comment](https://github.com/libp2p/rust-libp2p/pull/550#discussion_r224086310)